### PR TITLE
Setting X-PJAX-URL with a cross-domain URL should skip history.replaceState

### PR DESCRIPTION
**Pull request by mark-adams**
`window.history.replaceState()` triggers a DOMException if the URL does not share the same origin as the current page.

This PR skips the call to `window.history.replaceState()` if the new PJAX URL doesn't have the same origin.

### CODE REVIEW

**Comment by mislav** [comment]
Thanks. How did the `X-PJAX-URL` value get to point to another domain anyway?

**Comment by mark-adams** [comment]
@mislav In this case, the server sent the X-PJAX-URL pointed to another domain. In this particular case, the application serves some content from another domain instead of the primary domain and wants to ask PJAX to redirect its request to the other domain so it sets X-PJAX-URL.

**Comment by mislav** [comment]
Thanks for the explanation. I'm not convinced that we need to fix anything in pjax.

The idea of pjax is this:
* A resource is requested via ajax,
* Server returns partial HTML,
* HTML is injected into the layout and URL in the browser is updated as if navigation happened.

What pjax also handles:
* Doesn't handle clicks on links to another domain,
* Supports redirects via `X-PJAX-URL` as a workaround for the fact that XMLHttpRequest can't detect redirects,
* If the result HTML from the server is blank, pjax will perform a hard reload of the page. Users can take advantage of this feature in combination with `X-PJAX-URL` to perform hard redirects, if absolutely necessary.

Now, your use seems to be:
* The server sets `X-PJAX-URL` to another domain,
* But it also sends page content as a response. The combination of the two doesn't make sense.

If you want browser to follow an external link as a result of pjax navigation, then send a blank body in server response. But ideally, links that have a potential to redirect to other domains shouldn't be handled with pjax in the first place. Pjax is not ideal for these complex crossdomain scenarios.

**Comment by mark-adams** [comment]
Thanks for the information. I totally missed the PJAX handling of blank responses!
